processor/configuration/src/main/java/com/redhat/service/smartevents/processor/GatewaySecretsHandler.java

```diff
@@ -0,0 +1,46 @@
+package com.redhat.service.smartevents.processor;
+
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.redhat.service.smartevents.infra.models.gateways.Action;
+import com.redhat.service.smartevents.infra.models.gateways.Gateway;
+import com.redhat.service.smartevents.infra.models.gateways.Source;
+
+@ApplicationScoped
+public class GatewaySecretsHandler {
+
+    @Inject
+    ProcessorCatalogService processorCatalogService;
+
+    public static JsonNode emptyObjectNode() {
+        return new ObjectNode(JsonNodeFactory.instance);
+    }
+
+    public Action mask(Action action) {
+        List<String> passwordProps = processorCatalogService.getActionPasswordProperties(action.getType());
```

Comment:
"..password properties.." isn't very generic for a general "secrets" service.

Your tests show User Name and Token as two other types of secret. Did you want to consider changing the method name?